pkg/kubelet/dockertools/docker_test.go

```diff
@@ -698,10 +698,15 @@ func TestMakePortsAndBindings(t *testing.T) {
 			HostPort:      445,
 			Protocol:      "foobar",
 		},
+		{
+			ContainerPort: 443,
+			HostPort:      446,
+			Protocol:      "tcp",
+		},
 	}
 	exposedPorts, bindings := makePortsAndBindings(ports)
 	if len(ports) != len(exposedPorts) ||
-		len(ports) != len(bindings) {
+		len(ports) == len(bindings) {
```

Comment:
Both `len(bindings)` and `len(exposedPorts)` should equal to the the number of distinct container ports. `len(ports) == len(bindings)` doesn't do much here that the check below doesn't already do.

Below you check if any expected binding is there, but you never checked if there were unexpected (not repetitive) container->host port binds in the result. I see now you added a `default` case to reject unexpected bindings, so considered this comment resolved :)